File: rotatelist.java
class Solution{
  public ListNode rotateRight(ListNode head,int k){
    if(k==0 || head==null || head.next==null){return head;}
      ListNode last=head;
  }
  int length=1;
  while(last.next!=null){
    last=last.next;
    length++;
  }
  int rot=k%length;
  int skip=length-rot;
  ListNode newEnd=head;
  for(int i=0;i<skip-1;i++){
    newEnd=newEnd.next;
  }
  head=newEnd.next;
  newEnd.next=null;
  return head;
  
}

File: Rotate List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if (head == NULL)return head;
        if (head->next == NULL)return head;
        ListNode *temp = head, *new_head = NULL;
        int total_nodes = 1, cnt;
        while (temp->next!=NULL)
        {
            total_nodes += 1;
            temp = temp->next;
        }
        temp->next = head;
        k = k % total_nodes;
        cnt = (total_nodes - k );
        for(int i=0;i<cnt;i++)
        {
            temp = temp->next;
        }
        new_head = temp->next;
        temp->next = NULL;
        return new_head;
    }
};
